### PR TITLE
dialogflow-docs

### PR DESCRIPTION
**Pull request by jandonyada**

### CODE REVIEW

File: workbot/workbot-dialogflow-nlu.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+Back in your Workato account,
+- Go to App Connections
+- From the Application list, choose Google api.ai
+- Set the connection name
+- Paste the client access token (the one you copied from IssuesBot in api.ai. You copied that, right?)
+- Hit **Connect**
+
+![Google api.ai connection](/assets/images/workbot/workbot-dialogflow-nlu/google-apiai-connection.png)
+Your IssuesBot should now be connected to Workato.
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: should put a new line before this. May have UI bug where sentence sticks with the image.

File: workbot/workbot-dialogflow-nlu.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+Intents comprise of 4 parts:
+- **Training Phrases**
+- **Action & parameters**
+- **Response**
+- **Context**
+
+#### Training Phrases
+Training phrases teach the bot to extract parameters based off of sample phrases a user may say.
+
+![Training phrases](/assets/images/workbot/workbot-dialogflow-nlu/training-phrases.png)
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: I actually don't understand how training works here. 

Also the maybe you want to put `For 1`, `For 2`, ... in bullet points. From the Github preview I see them sticking together.

File: workbot/workbot-dialogflow-nlu.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+Lastly, for 5. `ineedhelp-issue-title-description-confirm`, we’ve trained the NLU to recognize the user’s confirmation when we ask if they’d like to create a Github ticket for their issue.
+
+#### Action & parameters
+Action & parameters define how to act on the values & parameters extracted from what the user says.
+
+![Action & parameters](/assets/images/workbot/workbot-dialogflow-nlu/action-and-parameters.png)
+
+**Parameter name**: name of the parameter we’re trying to extract
+
+**Entity**: Variables we want to elicit from the user.
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: what's `elicit`. Maybe try a more simple word?

File: workbot/workbot-dialogflow-nlu.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+**Value**: The actual value we want to use for the parameter. Prepend with \$ to use the resolved value e.g. `$title`, or omit it to use constants e.g. `100`.  
+
+At times, we may want to use what the user said in full (without resolving it to a value). To do so, we just append a '.original’ to the value e.g. `$title.original`.
+
+
+#### Responses
+After every intent prompt, you can define a response that helps to elicit information from the user in the follow-up intent. In this example, we reference previously resolved user parameter values as a positive feedback to the user that Workbot understood their input.
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: what's `elicit`. Maybe try a more simple word?

File: workbot/workbot-dialogflow-nlu.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+This is done by in the following format: `#context_name.parameter_name` but more on context later.
+
+#### Context
+Contexts are designed for passing on parameter values from previous intents.
+
+Input contexts act as a prerequisite for the intent to be matched; i.e. the intent will participate in matching only when *all* the contexts in the input context field are active.
+
+A good example would be in the final intent `ineedhelp-issue-title-description-confirm` where the bot asks for confirmation. The user need only reply with “ok”  - but the bot understands that its about creating the issue in Github. It knows this because the context from previous intents are being carried forward to this intent.
+
+![Context](/assets/images/workbot/workbot-dialogflow-nlu/context.png)
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: image is not showing up in preview

File: workbot/workbot-dialogflow-nlu.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+#### The final intent
+The final intent, `ineedhelp-issue-title-description-confirm` has 5 important parameters - 4 of which are responsible for triggering the recipe in Workato.
+- `response_type`: **command**
+- `Application`: **GitHub**
+- `Data`: **add**
+- `Action`: **data**
+
+`Param_query` is also an important parameter that references all parameter values resolved from the user in the current as well as previous intents.
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: I don't understand from this part onwards. Maybe a picture will make it easier to understand.

**Comment by jandonyada** [comment]
@zweedao addressed review comments, do take a look!

File: workbot/workbot-dialogflow-nlu.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+2. For `ineedhelp-issue`, we’ve trained the NLU with training phrases that allow it to recognize when a user is talking about a UI issue.
+
+3. For `ineedhelp-issue-title`, we’ve trained the NLU with training phrases that allow it to recognize 2 types of UI issues: images not loading and button misalignment.
+
+4. For `ineedhelp-issue-title-description`, we’ve trained the NLU with training phrases that allow it to recognize the user’s description of the issue.
+
+5. For `ineedhelp-issue-title-description-confirm`, we’ve trained the NLU with training phrases that allow it to recognize the user’s confirmation when we ask if they’d like to create a Github ticket for their issue.
+
+#### Action & parameters
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: This Action & parameters part is still pretty hard to understand.
I still can't distinguish between Parameter name, Entity, Value. Maybe a clear example with picture will make things easier.

File: workbot/workbot-dialogflow-nlu.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+
+#### Responses
+After every intent prompt, you can define a response that helps to collect information from the user in the follow-up intent. In this example, we reference previously resolved user parameter values as a positive feedback to the user that Workbot understood their input.
+
+![Response](/assets/images/workbot/workbot-dialogflow-nlu/response.png)
+
+This is done by in the following format: `#context_name.parameter_name` but more on context later.
+
+#### Context
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: still don't understand this part either :)

I think picture of how it looks in api.ai vs how it looks in Slack will help a lot.

File: workbot/workbot-dialogflow-nlu.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+Contexts are designed for passing on parameter values from previous intents.
+
+Input contexts act as a prerequisite for the intent to be matched; i.e. the intent will participate in matching only when *all* the contexts in the input context field are active.
+
+A good example would be in the final intent `ineedhelp-issue-title-description-confirm` where the bot asks for confirmation. The user need only reply with “ok”  - but the bot understands that its about creating the issue in Github. It knows this because the context from previous intents are being carried forward to this intent.
+
+![Context](/assets/images/workbot/workbot-dialogflow-nlu/context.png)
+
+#### The final intent
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: how is it `The final intent` linked to the 4 parts above (`Training Phrases`, `Action & parameters`, `Response`, `Context`)

What does `response_type`, `Application`, `Data`, `Action` do? How are they linked to a command / message in Workbot / Slack? Maybe pictures from Slack/Workbot vs api.ai will help to explain a lot.

**Comment by zweedao** [comment]
I read the whole article again and still don't understand very well how this whole thing works. 

Maybe it will be easier to understand if we have more pictures, which show a link between how it looks in api.ai vs how it looks in Slack.

So people look at this and see, "I setup this detail in api.ai and this is how it appears in Workbot"

File: workbot/workbot-dialogflow-nlu.md [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+Bot: Margherita pizza, yum!*
+
+In the example above, the bot knows that "Margherita" refers to the *pizza* (from the successful parameter extraction of the first user utterance 'I wanna order a pizza') as opposed to the cocktail of the same name.
+
+In IssueBot, you can see this at work in the `ineedhelp-issue-title-description` intent.
+
+![Context 2](/assets/images/workbot/workbot-dialogflow-nlu/context2.png)
+*Input & output contexts of `ineedhelp-issue-title-description`*
+
+This intent is matched only if *all* input contexts 'Title', 'Issue', and 'ineedhelp' are active. This ensures that when users reach this point of the conversation, 'Description' is the final parameter to extract. It's worth noting that the extracted parameter 'Description' is activated in the output context, so that the follow-up intent `ineedhelp-issue-title-description-confirm` can be matched to confirm all the parameters with the user.
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: Previously you say "trigger intent", but here you use "match intent", so reading this paragraph is a bit confusing. Also jumping right to `ineedhelp-issue-title-description` without explaining previous intents makes me quite lost.

Ideally, I think we should explain each intent 1 by 1 and show how they flow. This is for first-time user so need to be more step by step.

File: workbot/workbot-dialogflow-nlu.md [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+- Go to your Workbot connection and click on Edit connection (you may have to disconnect if the connection is active, so take care that its not being used in active recipes).
+- Under NLU provider, choose your Google api.ai connection.
+- Hit **Connect**
+
+![Workbot NLU provider](/assets/images/workbot/workbot-dialogflow-nlu/workbot-nlu-provider.png)
+
+Your Workbot should now be using Google api.ai as its NLU provider.
+
+## Testing it out
+Use this [recipe](https://www.workato.com/recipes/684354), and make the app connections above for the above Workbot connection and your Github connection.
```
Comment (zweedao) [inline workbot/workbot-dialogflow-nlu.md]: Do I need to start the recipe? When I type a message in Slack, should the Jobs report show anything?

File: connectors/dialogflow.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+
+13. After the Google Cloud SDK is up and running, execute the following command in terminal (replacing `<service-account-key-file.json>` with the path to the JSOn file you downloaded in step 10):
+```
+gcloud auth activate-service-account --key-file=<service-account-key-file.json>
+```
+15. Next, obtain the Client Access Token by executing the following command in terminal:
+```
+gcloud auth print-access-token
+```
+16. Use the the Client Access token in your Dialogflow connection in Workato.
```
Comment (eeshansim) [inline connectors/dialogflow.md]: 16. Use ~the~ the Client Access token in your Dialogflow connection in Workato.

File: connectors/dialogflow.md [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+# Google Dialogflow
+Dialogflow is a natural language understanding platform that makes it easy to design and integrate a conversational user interface into your mobile app, web application, device, bot, interactive voice response system, and so on. Using Dialogflow, you can provide new and engaging ways for users to interact with Workbot.
+
+When Workbot is connected with Dialogflow, all commands sent to Workbot will be processed by your connected Dialogflow bot's NLP (instead of Workbot's default NLP).
+
+## API version
+The Dialogflow connector uses [Dialogflow API v2](https://cloud.google.com/dialogflow/docs/reference/rest/v2-overview).
+
+## Workbot support for Dialogflow
+Currently, Workbot for Teams supports the use of Dialogflow on Workato.
```
Comment (eeshansim) [inline connectors/dialogflow.md]: Only Teams workbot? What about Slack workbot?

File: connectors/dialogflow.md [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+10. When the key is created, a download of the JSON file will start. Choose a location (remember the path to this location, you'll need it later!) to save it and confirm.
+> **Caution: You can only download this JSON file once, so make sure to save the file and keep it somewhere safe. If you lose this key or it becomes compromised, you can use the same process to create another.**
+
+11. Once complete, you'll see a pop up with a confirmation message. Click **Close**.
+![Create service account 3](/assets/images/connectors/dialogflow/create-key-3.png)
+
+12. Set up the Google Cloud SDK on your machine if you don't have it already. Follow the steps described here: https://cloud.google.com/sdk/docs/ (don't worry, it's quick and easy)
+
+13. After the Google Cloud SDK is up and running, execute the following command in terminal (replacing `<service-account-key-file.json>` with the path to the JSOn file you downloaded in step 10):
```
Comment (eeshansim) [inline connectors/dialogflow.md]: 13. After the Google Cloud SDK is up and running, execute the following command in terminal (replacing `<service-account-key-file.json>` with the path to the ~JSOn~ JSON file you downloaded in step ~10~ 7):

Comment (jandonyada) [reply connectors/dialogflow.md]: I renumbered the points, I think it's step 8 now. Please check.

File: connectors/dialogflow.md (hunk 13 above)
Comment (jandonyada) [reply connectors/dialogflow.md]: It's Slack workbot only, not Teams. Revised it.

**Pull request merged**